### PR TITLE
add `format_tbl()` function

### PR DESCRIPTION
**Pull request by MBattagl**
Added formatting function, which is currently named `format_tbl`. Since `format` is a base function, to call it `format` we'd need to have `descriptives` output be an S3 object, and add a format method. 

To get treatment counts in the column names, `descriptives()` now adds the counts as an attribute to the returned object.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -130,3 +130,92 @@ example_dat %>%
...
+    ## 8 age          ""     42.3 (22.1) 
+    ## 9 sugar_factor ""     0.49 (0.3)
+
+``` r
+univariate %>%
+  format_tbl()
+```
+
+    ## # A tibble: 12 x 3
+    ##    Variable     Label  Statistics
```
Comment (scheidec) [inline README.md]: Should the univariate table also have the sample size appended to the `Statistics` column name?

File: README.md [hunk 2]
```diff
@@ -130,3 +130,92 @@ example_dat %>%
...
+
+``` r
+bivariate %>%
+  format_tbl()
+```
+
+    ## # A tibble: 12 x 5
+    ##    Variable     Label  `candy (N=257)` `ice cream (N=243)` `P Value`
+    ##    <chr>        <chr>  <chr>           <chr>               <chr>    
+    ##  1 age          ""     42.26 (22.62)   42.39 (21.61)       0.946
```
Comment (scheidec) [inline README.md]: Looks like the cat/cont ordering doesn't hold after running `format_tbl()`.  Can that be preserved?

**Review by scheidec — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (MBattagl) [reply README.md]: It can if we want it to. In practice, I only add N's to the treatment levels.

Comment (scheidec) [reply README.md]: I think it's worth it to add that in for the summary of the sample to be complete.

File: README.md (hunk 2 above)
Comment (MBattagl) [reply README.md]: Yeah I noticed this too, and agree that it'd be nice to preserve the ordering from the earlier table. The implementation with `dplyr::group_split()` is elegant, but may need to switch to `base::split()`. I'll work on getting this behavior

**Comment by MBattagl** [comment]
Updated. To preserve the ordering all I needed to do was convert `Variable` to a factor myself prior to passing it into `group_by`.

**Review by scheidec — APPROVED** [review]

**Pull request merged**